### PR TITLE
[OSD-6207] Documentation Improvements

### PR DESCRIPTION
**Pull request by dkeohane**
[OSD-6207](https://issues.redhat.com/browse/OSD-6207)
- Added [Go Report Card](https://goreportcard.com/) Badge to README
- Added Go Doc Badge to README
- Added docs README, Glossary, Debugging pages
- Updated Installation docs & moved prerequisites into their own file.
- Grouped all CRs and Controllers by their type (AccountPool, Account etc) instead of their function (CR vs Controllers) as I felt it's easier to read this way

### CODE REVIEW

File: docs/1.0-Installation.md [hunk 1]
```diff
@@ -1,20 +1,37 @@
...
+    - [1.5.2 Cluster Mode](#152-cluster-mode)
+
 ### 1.0.1 - Glossary
 
 Terms used in this guide and what they mean for this guide:
 
 * Payer Account - AWS Organizations give you a "unified bill" so to speak under a single account. Sub accounts are created under this "Payer Account", which is the highest-level account in this organization. SREs all should also have an IAM user with credentials on each payer account, which has AssumeRole Permissions as administrator into each of these sub-accounts.
-* Jump Role - Using AWS Assume-Role Chaining, a Jump Role is an intermediary role that sits between a given IAM user and a customer's access role, providing a single point of entry.
+* Jump Role - Using AWS Assume-Role Chaining, a `Jump Role` is an intermediary role that sits between a given IAM user and a customer's `Access Role`, providing a single point of entry.
 * Access Role - A role inside the cluster's account that we assume to perform tasks.
```
Comment (iamkirkbater) [inline docs/1.0-Installation.md:26]: Can we move this section to the glossary page?

File: docs/1.0-Installation.md [hunk 2]
```diff
@@ -63,19 +80,34 @@ Alternatively, if you have the credentials stored in your `~/.aws/credentials` f
...
+- **Account 1:**
+  - This Account will be your Jump Host
+  - It will have a newly created `Policy` to `Assume Role`
+  - It will have a newly created `Role` for its `Jump Role`
+    - The new `Policy` created, `Assume Role`, will be attached to this role
+    - The `Trust Relationship` for this role will be updated to trust the `Payer Account`
+- **Account 2:**
+  - This Account is the one we will gain access to via the `Access Role`
+  - It will have a new role
+    - The Trust Relationship for this role will be updated to trust Account 1 Jump Role ARN
```
Comment (iamkirkbater) [inline docs/1.0-Installation.md:102]: Nice clarification!

File: docs/1.0-Glossary.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+
+# 1.0 Glossary
```
Comment (iamkirkbater) [inline docs/1.0-Glossary.md:2]: Can we change the name of this file and this section to be `0.1-Glossary`?

File: docs/1.0-Local-Development.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+# Setup your Local Development Environment
```
Comment (iamkirkbater) [inline docs/1.0-Local-Development.md:1]: Can we rename this file to be `1.1-Local-Development.md`?

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Documentation
```
Comment (iamkirkbater) [inline docs/index.md:1]: Let's remove this file in favor of the work that @drpaneas is doing in #592.

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Documentation
```
Comment (iamkirkbater) [inline docs/README.md:1]: Let's remove this file in favor of the work that @drpaneas is doing in #592.

File: mkdocs.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+site_name: AWS Account Operator
```
Comment (iamkirkbater) [inline mkdocs.yml:1]: Let's remove this file in favor of the work that @drpaneas is doing in #592.

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
Overall I really like the changes.  There are a few minor bits, but if we make those changes we can merge this along with #592 and have some really nice docs :)

File: docs/1.0-Installation.md (hunk 1 above)
Comment (dkeohane) [reply docs/1.0-Installation.md]: Done

File: docs/1.0-Installation.md (hunk 2 above)
Comment (dkeohane) [reply docs/1.0-Installation.md]: Thanks! :)

File: docs/1.1-InstallationPrerequisites.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+Once you have the two accounts assigned to you, you can access their consoles via [osdctl](https://github.com/openshift/osdctl/#aws-account-console-url-generate):
+```
+osdctl account console -p $AWSPROFILE -i $ACCOUNT_ID
+```
+
+### 1.1.3 STS Roles
+
+To run the operator tests for [STS](0.1-Glossary.md#sts) mode you will need to create two roles &mdash; one in each of the AWS accounts you have been assigned. The name of these roles doesn't matter when you create it, however you will need to know the ARN of both in order to configure the operator to use them. By the end of these steps your two AWS accounts will look as follows:
+- **Account 1:**
```
Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:80]: I believe the diagram is needed here as well

Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:80]:
```suggestion
- **Account 1 (Jump Host)** 
```

File: docs/1.1-InstallationPrerequisites.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+  - This Account will be your Jump Host
+  - It will have a newly created `Policy` to `Assume Role`
+  - It will have a newly created `Role` for its `Jump Role`
+    - The new `Policy` created, `Assume Role`, will be attached to this role
+    - The `Trust Relationship` for this role will be updated to trust the `Payer Account`
+- **Account 2:**
```
Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:86]:
```suggestion
- **Account 2 (Target) **
```

File: docs/1.1-InstallationPrerequisites.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+  - This Account is the one we will gain access to via the `Access Role`
+  - It will have a new role
+    - The Trust Relationship for this role will be updated to trust Account 1 Jump Role ARN
+
+#### 1.1.3.1 - Jump Role
+
+The `Jump Role` will be a simulation of the role that we use as a bastion of sorts. For the STS Architecture, we have an IAM user that assumes a specific role in a specific account, and then using that role they can assume role into the cluster's account in order to run operations as necessary.
```
Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:93]:
```
The `Jump Role` will be a simulation of the role that we use as a bastion of sorts.
```
Can you rephrase this sentence a bit so it becomes more generic?

File: docs/1.1-InstallationPrerequisites.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+You will need to create this `Jump Role` in the first osd-staging aws account assigned to you. You will need to add your `Payer Account` IAM user credentials as the Principal in the `Trust Relationship`. Ensure this is the same user used to create the AWS Account Operator credentials.
```
Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:95]:
```suggestion
You will need to create this `Jump Role` in the first osd-staging AWS account assigned to you. You will need to add your `Payer Account` IAM user credentials as the Principal in the `Trust Relationship`. Ensure this is the same user used to create the AWS Account Operator credentials.
```

File: docs/1.1-InstallationPrerequisites.md [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+        "ec2:DescribeInstanceStatus",
+        "ec2:CreateTags",
+        "ec2:DeleteTags"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+```
+**Note:** You will need to put the ARN for this `Access Role` into your environment in order for this to be used to create accountclaims within the make tests.
```
Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:177]: you can mention the env var here

File: docs/1.1-InstallationPrerequisites.md [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+
+### 1.1.4 Config Map
+
+The operator needs a ConfigMap that has the following fields configured:
+
+* `account-limit`: The soft limit of AWS accounts which is the number compared against when creating new accounts
+* `base`: Base [OU](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_ous.html) ID to place accounts in when claimed
+* `root`: Root [OU](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_ous.html) ID to create new OUs under
```
Comment (dastergon) [inline docs/1.1-InstallationPrerequisites.md:185]: sts-jump-role is missing

**Review by dastergon — CHANGES_REQUESTED** [review]

File: docs/0.1-Glossary.md [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+### Payer Account
+[AWS Organizations](https://aws.amazon.com/organizations/) give you a "unified bill" so to speak under a single account. Sub accounts are created under this "Payer Account", which is the highest-level account in this organization. SREs all should also have an IAM user with credentials on each payer account, which has AssumeRole Permissions as administrator into each of these sub-accounts.
+
+### Red Hat Managed Cluster 
+Standard OpenShift Dedicated clusters are deployed into their own cloud infrastructure accounts, each owned by Red Hat. Red Hat is responsible for this account. From this account, only Red Hat managed clusters will be visible, in other words, CCS/BYOC clusters will not.
+
+### CCS / BYOC
+CCS *(Customer Cloud Subscriptions)* or BYOC *(Bring your Own Cluster)* clusters are those which belong to customers’ personal AWS accounts, as opposed to being owned by RedHat. Therefore, for CCS we are provisioning a cluster in an account provided by the customer. For CCS clusters, we are going to find multiple clusters per account.
+
+### Trust Relationship
```
[the remark this answers is not in the document]
Comment (dkeohane) [reply docs/0.1-Glossary.md:22]: I added the description from AWS instead

File: docs/0.1-Glossary.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+### Jump Role
+Using AWS Assume-Role Chaining, a Jump Role is an intermediary role that sits between a given IAM user and a customer's access role, providing a single point of entry.
```
[the remark this answers is not in the document]
Comment (dkeohane) [reply docs/0.1-Glossary.md]: done

**Review by drpaneas — APPROVED** [review]
/lgtm

**Review by dastergon — APPROVED** [review]
/lgtm

File: docs/0.1-Glossary.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+This glosary is intended to catalogue and define all terms used in our documentation which may not be evident to new users of the operator.
+
+By extension of this glossary, we may also use terms from the following glossaries:
+* [OLM](https://docs.openshift.com/container-platform/4.7/operators/understanding/olm-common-terms.html)
+* [Kubernetes](https://kubernetes.io/docs/reference/glossary/?fundamental=true)
+
+
+## Terms
+### Payer Account
+[AWS Organizations](https://aws.amazon.com/organizations/) give you a "unified bill" so to speak under a single account. Sub accounts are created under this "Payer Account", which is the highest-level account in this organization. SREs all should also have an IAM user with credentials on each payer account, which has AssumeRole Permissions as administrator into each of these sub-accounts.
```
Comment (rogbas) [inline docs/0.1-Glossary.md:12]: `SREs all should also have an IAM user with credentials on each payer account, which has AssumeRole Permissions as administrator into each of these sub-accounts.`

I'm not sure I'd refer to SREs here since this is a public-facing doc.

File: docs/0.1-Glossary.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+
+### Two Cluster Types
+We support 2 types of clusters:
+- **Red Hat Managed Cluster:** Standard OpenShift Dedicated clusters are deployed into their own cloud infrastructure accounts, each owned by Red Hat. Red Hat is responsible for this account. From this account, only Red Hat managed clusters will be visible, in other words, CCS/BYOC clusters will not.
+- **CCS / BYOC:** CCS *(Customer Cloud Subscriptions)* or BYOC *(Bring your Own Cluster)* clusters are those which belong to customers’ personal AWS accounts, as opposed to being owned by Red Hat. Therefore, for CCS we are provisioning a cluster in an account provided by the customer. For CCS clusters, we are going to find multiple clusters per account.
```
Comment (rogbas) [inline docs/0.1-Glossary.md:17]: I would be good to link to our CCS doc: https://www.openshift.com/dedicated/ccs

File: docs/0.1-Glossary.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+### STS
+[AWS Security Token Service (AWS STS)](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) is a web service that enables you to request temporary, limited-privilege credentials for AWS Identity and Access Management (IAM) users or for users that you authenticate (federated users).
+
+### Trust Relationship
+Trust relationships/Trust Policies are then configured between the IAM users and the IAM roles. This policy defines which principals can assume the role, and under which conditions. This is sometimes referred to as a resource-based policy for the IAM role.
+
+### Assume Role
+Returns a set of temporary security credentials that you can use to access AWS resources that you might not normally have access to. These temporary credentials consist of an access key ID, a secret access key, and a security token. Typically, you use AssumeRole within your account or for cross-account access.
```
Comment (rogbas) [inline docs/0.1-Glossary.md:26]: We could link the respective AWS doc here. This [1] or something similar.

[1] https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html

**Comment by iamkirkbater** [comment]
Overall LGTM.  Can you rebase/squash to a single commit and then I'll give the :+1: and merge it in?

**Comment by rogbas** [comment]
thanks for that @dkeohane overall it's looking good, left some minor comments.
